File: 프로그래머스/unrated/150370. 개인정보 수집 유효기간/개인정보 수집 유효기간.py
def solution(today, terms, privacies):
    answer = []
    term_dict = {}
    for term in terms:
        a, b = term.split()
        term_dict[a] = int(b)
        y2, m2, d2 = map(int, today.split("."))
    for i in range(len(privacies)):
        tmp1, tmp2 = privacies[i].split()
        y1, m1, d1 = map(int, tmp1.split("."))
        k1 = y1*12*28 + (m1+term_dict[tmp2])*28 + d1 - 1
        k2 = y2*12*28 + m2*28 + d2
        
        if k2 > k1:
            answer.append(i+1)
                
    answer.sort()
    
    return answer

File: 프로그래머스/1/150370. 개인정보 수집 유효기간/개인정보 수집 유효기간.java
import java.time.LocalDate;
import java.util.ArrayList;
import java.util.HashMap;
import java.util.List;
import java.util.Map;

class Solution {
    public int[] solution(String today, String[] terms, String[] privacies) {
        int[] answer = {};

        Map<String, Integer> map = new HashMap<>();
        for (String term : terms) {
            String[] s = term.split(" ");
            map.put(s[0], Integer.parseInt(s[1]));
        }

        String[] split = today.split("\\.");
        int year = Integer.parseInt(split[0]);
        int month = Integer.parseInt(split[1]);
        int day = Integer.parseInt(split[2]);
        LocalDate todayDate = LocalDate.of(year, month, day);

        List<Integer> list = new ArrayList<>();
        for (int i = 0; i < privacies.length; i++) {
            String privacy = privacies[i];
            String[] s = privacy.split(" ");
            String[] ymd = s[0].split("\\.");
            int y = Integer.parseInt(ymd[0]);
            int m = Integer.parseInt(ymd[1]);
            int d = Integer.parseInt(ymd[2]);
            LocalDate date = LocalDate.of(y, m, d);

            if (!date.plusMonths(map.get(s[1])).isAfter(todayDate)) {
                list.add(i + 1);
            }
        }

        answer = list.stream().mapToInt(i -> i).toArray();

        return answer;
    }
}